### PR TITLE
Upgrade to Babel 6

### PR DESCRIPTION
**Pull request by kevva**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
   },
   "scripts": {
     "test": "xo && npm run prepublish && nyc ava",
-    "prepublish": "babel lib --out-dir=dist --optional=runtime",
+    "prepublish": "babel lib --out-dir=dist --presets es2015 --plugins transform-async-to-generator,transform-runtime",
```
Comment (SamVerschueren) [inline package.json]: Maybe it's cleaner to just add a babel configuration property to `package.json`?

```json
{
  "babel": {
    "plugins": [
      "transform-async-to-generator",
      "transform-runtime"
    ],
    "presets": [
      "es2015"
    ]
  }
}
```

File: package.json [hunk 2]
```diff
@@ -66,7 +66,6 @@
   "dependencies": {
     "array-differ": "^1.0.0",
     "array-uniq": "^1.0.2",
-    "babel-runtime": "^5.8.29",
```
Comment (SamVerschueren) [inline package.json]: Is it safe to remove this one? Just wondering, not a babel expert at all :).

Comment (kevva) [inline package.json]: Oh, probably not.

File: package.json (hunk 1 above)
Comment (kevva) [inline package.json]: Probably, I just kept it as is since there aren't that many plugins.

Comment (SamVerschueren) [inline package.json]: I'm ok either way :).

Comment (sindresorhus) [inline package.json]: :+1: package.json

File: package.json (hunk 2 above)
Comment (sindresorhus) [inline package.json]: No, still needed. It's for stuff that can't be statically transformed, like async functions.

**Comment by sindresorhus** [comment]
To support `nyc`:

> Source maps are used to map coverage information back to the appropriate lines of the pre-transpiled code. You'll have to configure your custom require hook to inline the source map in the transpiled code. **For Babel that means setting the sourceMaps option to inline.** - https://github.com/bcoe/nyc#support-for-custom-require-hooks-babel-es2015

**Comment by sindresorhus** [comment]
Welcome back @kevva! 😎✨

**Comment by SamVerschueren** [comment]
> Welcome back @kevva! 😎✨

2nd that :)

**Comment by kevva** [comment]
> Welcome back @kevva! 😎✨

Yay!

**Comment by SamVerschueren** [comment]
Looked into the failing tests. It seems that running [this](https://github.com/sindresorhus/pageres/blob/babel-6/test/test.js#L143-L146) test serially fixes the failing tests which is quite odd.

I tried to track it down when `serially` is not defined and it seems that there is something going on in `screenshot-stream`.

```js
var stream = cp.stdout.pipe(base64Stream.decode());

process.stderr.setMaxListeners(0);

cp.stderr.setEncoding('utf8');
cp.stderr.on('data', function (data) {
	stream.emit('error', new Error('foo bar'));
});

return stream
```

results in

```
Uncaught Exception: test/test.js
  Error: foo bar
```

Tried surrounding everything with `try-catch` etc but I'm out of ideas at the moment.

**Comment by kevva** [comment]
Strange indeed, and I haven't been able to reproduce it locally which makes it hard to debug.

**Comment by SamVerschueren** [comment]
I have actually. Tried `rm -rf node_modules && npm install`?

It seems that it has something to do with `nyc`. If I change the test command to `xo && npm run prepublish && ava` it passes. If I add `nyc`, it fails.

**Comment by SamVerschueren** [comment]
I'll push that change to verify my assumption on travis.

**Comment by kevva** [comment]
> I have actually. Tried rm -rf node_modules && npm install?

Yup.

**Comment by SamVerschueren** [comment]
It looks like `nyc` is the problem here.

**Comment by sindresorhus** [comment]
https://github.com/bcoe/nyc/issues/190 might have been the issue. Can you try adding back `nyc`?

**Comment by kevva** [comment]
Apparently not :disappointed:

**Comment by sindresorhus** [comment]
Hmm, let's just `test.skip` the failing test for now.

**Comment by kevva** [comment]
Yup!

File: package.json [hunk 3]
```diff
@@ -85,19 +85,32 @@
...
     "sinon": "^1.17.2",
     "xo": "*"
   },
+  "babel": {
+    "plugins": [
+      "transform-async-to-generator",
+      "transform-runtime"
+    ],
+    "presets": [
+      "es2015"
```
Comment (sindresorhus) [inline package.json]: We use async function so needs the stage-3 preset too.

**Comment by sindresorhus** [comment]
Woot! 🎉

**Comment by kevva** [comment]
Wooooooot :cake:

**Comment by SamVerschueren** [comment]
Nice work guys!

**Pull request merged**